azurerm/resource_arm_container_group_test.go

```diff
@@ -11,11 +11,78 @@ import (
+					resource.TestCheckResourceAttr(resourceName, "image_registry_credential.0.server", "hub.docker.com"),
+					resource.TestCheckResourceAttr(resourceName, "image_registry_credential.0.username", "yourusername"),
+					resource.TestCheckResourceAttr(resourceName, "image_registry_credential.0.password", "yourpassword"),
+					resource.TestCheckResourceAttr(resourceName, "image_registry_credential.1.server", "mine.acr.io"),
+					resource.TestCheckResourceAttr(resourceName, "image_registry_credential.1.username", "acrusername"),
+					resource.TestCheckResourceAttr(resourceName, "image_registry_credential.1.password", "acrpassword"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMContainerGroup_imageRegistryCredentialsUpdate(t *testing.T) {
+	resourceName := "azurerm_container_group.test"
+	ri := acctest.RandInt()
+
+	config := testAccAzureRMContainerGroup_imageRegistryCredentials(ri, testLocation())
+	updated := testAccAzureRMContainerGroup_imageRegistryCredentialsUpdated(ri, testLocation())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMContainerGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMContainerGroupExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "image_registry_credential.#", "2"),
+					resource.TestCheckResourceAttr(resourceName, "image_registry_credential.0.server", "hub.docker.com"),
+					resource.TestCheckResourceAttr(resourceName, "image_registry_credential.0.username", "yourusername"),
+					resource.TestCheckResourceAttr(resourceName, "image_registry_credential.0.password", "yourpassword"),
+					resource.TestCheckResourceAttr(resourceName, "image_registry_credential.1.server", "mine.acr.io"),
+					resource.TestCheckResourceAttr(resourceName, "image_registry_credential.1.username", "acrusername"),
+					resource.TestCheckResourceAttr(resourceName, "image_registry_credential.1.password", "acrpassword"),
+				),
+			},
+			{
+				Config: updated,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMContainerGroupExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "image_registry_credential.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "image_registry_credential.0.server", "hub.docker.com"),
+					resource.TestCheckResourceAttr(resourceName, "image_registry_credential.0.username", "updatedusername"),
+					resource.TestCheckResourceAttr(resourceName, "image_registry_credential.0.password", "updatedpassword"),
+				),
+			},
+		},
+	})
+}
+
 func TestAccAzureRMContainerGroup_linuxBasic(t *testing.T) {
 	resourceName := "azurerm_container_group.test"
 	ri := acctest.RandInt()
 
-	config := testAccAzureRMContainerGroup_linuxBasic(ri, testLocation())
+	config := testAccAzureRMContainerGroup_imageRegistryCredentials(ri, testLocation())
```

Comment:
the tests are failing since this has been updated but the counts are incorrect; going to push a test to revert this to `TestAccAzureRMContainerGroup_linuxBasic`